### PR TITLE
Publish android artifacts with publish to s3 gradle plugin

### PR DESCRIPTION
**Pull request by oguzkocer**
## Description
Update `react-native-bridge` and `react-native-aztec` projects to use [publish-to-s3-gradle-plugin](https://github.com/Automattic/publish-to-s3-gradle-plugin). This plugin allows us to simplify the logic in the clients by adding a `prepareToPublishToS3` Gradle task which will set the version name according to the provided parameters and verify that version is not yet published.

_Please note that this PR should not be merged before gutenberg-mobile PR (which I'll create and link here soon ™️ ) is approved and ready to be merged as this PR contains breaking changes. At this point, I am only looking for code review and will ask for it to be tested in the `gutenberg-mobile` PR instead. I'll also add some line comments explaining the reasoning behind some changes._ cc @jkmassel 

## How has this been tested?

* Manually tested `react-native-bridge` & `react-native-aztec` libraries and verified that they build in my local environment
* Implemented Buildkite integration for these projects in `gutenberg-mobile` repository and verified that artifacts were published correctly
* Used the published artifacts in WPAndroid and verified that the editor worked
- [ ] `react-native-editor` is still yet to be tested although I don't expect it to be impacted by these changes.

## Types of changes

Build file changes for `react-native-bridge` & `react-native-aztec` Android libraries.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/react-native-aztec/android/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buildscript {
     ext {
         gradlePluginVersion = '4.0.2'
-        kotlinVersion = '1.3.61'
+        kotlinVersion = '1.5.20'
```
Comment (oguzkocer) [inline packages/react-native-aztec/android/build.gradle:4]: Previous kotlin version was resulting in some build failures, but unfortunately I can't remember the details of the failure. The upgrade didn't cause any issues, so I don't think we need to be concerned about this version upgrade.

File: packages/react-native-aztec/android/build.gradle [hunk 2]
```diff
@@ -12,10 +12,16 @@ buildscript {
         wordpressUtilsVersion = '1.22'
         espressoVersion = '3.0.1'
         aztecVersion = 'v1.3.45'
-        willPublishReactNativeAztecBinary = properties["publishReactNativeAztecVersion"]?.trim() as boolean
+        willPublishReactNativeAztecBinary = properties["willPublishReactNativeAztecBinary"]?.toBoolean() ?: false
```
Comment (oguzkocer) [inline packages/react-native-aztec/android/build.gradle:15]: I am still not particularly happy about this naming (and the similar one in `react-native-bridge`), but I can't think of a better name that communicates what it's about.

File: packages/react-native-aztec/android/build.gradle [hunk 3]
```diff
@@ -24,13 +30,14 @@ buildscript {
...
         classpath 'com.github.dcendents:android-maven-gradle-plugin:2.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
+        classpath 'com.automattic.android:publish-to-s3:0.6.0'
     }
 }
 
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: 'com.github.dcendents.android-maven'
-apply plugin: 'maven-publish'
```
Comment (oguzkocer) [inline packages/react-native-aztec/android/build.gradle:33]: `maven-publish` plugin will be applied by `publish-to-s3` gradle plugin.

File: packages/react-native-aztec/android/build.gradle [hunk 4]
```diff
@@ -116,34 +123,27 @@ dependencies {
...
-            }
-            repositories {
-                maven {
-                    url "s3://a8c-libs.s3.amazonaws.com/android"
-                    credentials(AwsCredentials) {
-                        accessKey project.hasProperty('awsAccessKey') ? project.properties['awsAccessKey'] : System.getenv('AWS_ACCESS_KEY')
-                        secretKey project.hasProperty('awsSecretKey') ? project.properties['awsSecretKey'] : System.getenv('AWS_SECRET_KEY')
-                    }
-                }
+                addDependenciesToPom(pom)
```
Comment (oguzkocer) [inline packages/react-native-aztec/android/build.gradle:136]: I'd like to remove this and the `artifact bundleReleaseAar` property and instead utilize `from.release` (which should add the necessary dependencies), but my initial test failed and I think it deserves its own PR, so I am leaving it as is.

File: packages/react-native-aztec/android/build.gradle [hunk 5]
```diff
@@ -116,34 +123,27 @@ dependencies {
...
             }
         }
-    }
+   }
+}
+
+publishToS3Plugin {
+    mavenPublishGroupId 'org.wordpress-mobile.gutenberg-mobile'
+    mavenPublishArtifactId 'react-native-aztec'
```
Comment (oguzkocer) [inline packages/react-native-aztec/android/build.gradle:144]: These properties are used to check if a version is already published to S3. It should match the information in the `publish` block.

I am going to re-visit this later and try to get rid of it - or at least extract the values so we don't have to reference them twice.

File: packages/react-native-bridge/android/publish-aztec-and-bridge.sh [hunk 6]
```diff
@@ -1,53 +0,0 @@
-#!/bin/bash
```
Comment (oguzkocer) [inline packages/react-native-bridge/android/publish-aztec-and-bridge.sh:1]: This script is no longer necessary because:
1. We are moving the logic the `gutenberg-mobile` repository
2. Some of the logic is covered by `publish-to-s3` plugin
3. We are simplifying this process to address some developer frustrations

File: packages/react-native-bridge/android/settings.gradle [hunk 7]
```diff
@@ -3,31 +3,35 @@ rootProject.name = '@wordpress_react-native-bridge'
...
-include ':react-native-screens'
-project(':react-native-screens').projectDir = new File(rootProject.projectDir, '../../../node_modules/react-native-screens/android')
-include ':react-native-safe-area-context'
-project(':react-native-safe-area-context').projectDir = new File(rootProject.projectDir, '../../../node_modules/react-native-safe-area-context/android')
-include ':react-native-reanimated'
-project(':react-native-reanimated').projectDir = new File(rootProject.projectDir, '../../../node_modules/react-native-reanimated/android')
-include ':react-native-prompt-android'
-project(':react-native-prompt-android').projectDir = new File(rootProject.projectDir, '../../../node_modules/react-native-prompt-android/android')
+
+if (hasProperty("willPublishReactNativeBridgeBinary")) {
```
Comment (oguzkocer) [inline packages/react-native-bridge/android/settings.gradle:9]: This `if/else` block is necessary to be able to configure the project when the `node_modules` folder is missing from the project. With Buildkite integration, we are separating the Android artifact publishing task from the Android resource bundling, so `node_modules` will no longer be available and the project is expected to only work with the `index.android.bundle` file.

**Comment by jkmassel** [comment]
Oguz and I went over this together yesterday and it LGTM!

**Review by mchowning — APPROVED** [review]
Thanks for this @oguzkocer! Also thanks for all the explanatory comments on this PR! 😄
